gematria/granite/python/token_graph_builder_model.py

```diff
@@ -340,3 +358,76 @@ def _create_graph_network_modules(
             residual_connection=options.EnableFeature.BY_FLAG,
         ),
     )
+
+  def _make_batch_feed_dict(self) -> model_base.FeedDict:
+    feed_dict = super()._make_batch_feed_dict()
+    to_feed_instruction_annotations = None
+    if len(self._batch_graph_builder.instruction_annotations) == 0:
```

Comment:
Two comments:
1. I suspect that each access to `graph_builder.instruction_annotations` creates a new copy of the map/vectors. That can be very expensive, and it's best avoided.
2. The "canonical" way to check whether a container is empty in Python is to use the implicit boolean conversion.

```
instruction_annotations = self._batch_graph_builder.instruction_annotations
if not instruction_annotations:
  ...
```